src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage.Accessor.cs

```diff
@@ -255,9 +261,15 @@ private bool WriteStream(TKey key, Stream stream, Checksum? checksum, Cancellati
                 // Have to run the checksum reading in a transaction.  This is necessary as blob reading outside a
                 // transaction is not safe to do with the sqlite API.  It may produce corrupt bits if another thread is
                 // writing to the blob.
-                return connection.RunInTransaction(
+                var (stream, exception) = connection.RunInTransaction(
                     static t => t.connection.ReadChecksum_MustRunInTransaction(t.database, t.self.Table, t.rowId),
-                    (self: this, connection, database, rowId));
+                    (self: this, connection, database, rowId),
+                    throwOnSqlException: true);
+
+                // we should never have gotten a SqlException while reading since we passed throwOnSqlException: true above.
+                Contract.ThrowIfTrue(exception != null);
+
+                return stream;
```

Comment:
reads should still nto fail, so we still want to report exceptions if they do so we can investigate.

Reply:
Are there no error codes we're still okay with that are recoverable?